### ISSUE TITLE
Provide a property in the frame object that indicates the item index for that frame

### ISSUE THREAD

**Issue by Dan503** [pixfabrik/driftory-viewer#26]
I'm creating a navigation pane that allows the user to quickly jump between comic frames. 

The navigation is a bunch of little thumbnail images that the user can click on to go to the pictured frame.

At the moment, I don't have any easy way of determining which image is being displayed in each frame.

I think if the frame object included an `itemIndex` property specifying which image item is being displayed in that frame, it would help me a lot with determining which image to display in the thumbnail.

I'm thinking that I will proably be able to use the x and y coordinates to figure out which part of the image to display, the important thing I'm missing though is knowing which image needs to be displayed in the first place.

Note something to think about: what hapens if someone places a frame out in the middle of nowhere? Also, need to think about the "don't show until frame" option.

**Comment by iangilman** [comment]
Interesting! Okay, I've provided a mechanism for this in #27. Take a look at the "list images" feature in the demo to see how it works. I think it may take care of the "hide until frame" images properly even, though it might not be as robust on that front as it could be. Let me know if you encounter any weirdness. I'll make a note to make that handling more robust, but otherwise I think this is taken care of.

**Comment by Dan503** [comment]
@iangilman Not exactly what I had in mind but I see what you did in demo.ts (line 111).

I'll see if I can use that technique to create my own list of thumbnails.

**Comment by Dan503** [comment]
I'm going to close this issue off now because I was able to get it working based on the example you gave in the demo. 🙂

If you're interested, this is what the code looks like:

```js
const round = (dp = 2) => Math.round((dp + Number.EPSILON) * 100) / 100
const frames = comicData.comic.body.frames || comicData.comic.body.items

const thumbnailBgStyles = frames.map((frameData, frameIndex) => {
	const frame = driftoryViewer?.getFrame(frameIndex)
	if (frame?.images.length) {
		const key = JSON.stringify(frameData)
		const thumbnailWidth = 200
		const frameBounds = frame.bounds
		const shrinkMultiplier = thumbnailWidth / frameBounds.width
		const frameHeight = frameBounds.height * shrinkMultiplier

		return frame.images
			.sort((a, b) => {
				if (a.hideUntilFrame === undefined) {
					return 1
				}
				if (b.hideUntilFrame === undefined) {
					return -1
				}
				if (a.hideUntilFrame < b.hideUntilFrame) {
					return 1
				}
				if (b.hideUntilFrame > a.hideUntilFrame) {
					return -1
				}
				return 0
			})
			.map(frameImage => {
				const imgBounds = frameImage.bounds as Frame
				const imgHeight = round(imgBounds.height * shrinkMultiplier)
				const imgWidth = round(imgBounds.width * shrinkMultiplier)
				const imgX = round(
					(imgBounds.x - frameBounds.x) * shrinkMultiplier
				)
				const imgY = round(
					(imgBounds.y - frameBounds.y) * shrinkMultiplier
				)
				// I don't want to download the images until I have to
				const imgUrl = isNavOpen ? frameImage.url : ''
				const bgString = `${imgX}px ${imgY}px / ${imgWidth}px ${imgHeight}px no-repeat url(${imgUrl})`
				return bgString
			})
	}
})
```

![sample image of how the thumbnails look when used.](https://user-images.githubusercontent.com/10610368/99892619-b4025600-2cca-11eb-9fda-edb971a08a15.png)

I also tried this second style where I have the text boxes as their own separate image laid on top of the scene image.

![another sample of how the thumbnails can look](https://user-images.githubusercontent.com/10610368/99895820-9fc85400-2cde-11eb-9e43-5efbad7bb626.png)

**Comment by iangilman** [comment]
Cool! I like how you are taking just the right part of the image. That might be interesting to add to the `getFrame` functionality.

BTW, you shouldn't have to do any sorting… It's already sorted by which image takes up the largest area of the frame, and "hide until frame" images that are not applicable are already filtered out. At least, that's the intention of the code! Let me know if it's not doing that properly.

**Comment by Dan503** [comment]
> BTW, you shouldn't have to do any sorting

I had issues with the way the images were getting sorted. Since I am using background image, the image that is first in the list is the image that will appear on top. This makes sense when you write CSS manually.

```css
.bg {
  background:
    top-layer,
    middle-layer,
    bottom-layer;
}
```

The current order puts the item that is in the top layer last in the list. At first I tried just reversing the order but I think I ran into an issue where values with no `hideUntilFrame` value were placed in the incorrect order compared to values that do have a `hideUntilFrame` value.

It would probably be useful to build this into the API directly though as a `getFrameImagesAsBackgroundStyle(frameWidth)` so users can do this common use case more easily without having to write all the code themselves. I'll create a new issue for that.
